### DISCUSSION TITLE
Close the window with multiple tabs to **easily** find and restore it later

### DISCUSSION THREAD

**Discussion by dimitry12** [mbnuqw/sidebery#1348] [Q&A] [answered]
I use Sidebery and "Tab Count in Window Title" extensions in Firefox on MacOS. Big fan of both extensions: running dozens of windows with thousands of tabs across them. In my alt-tab/window-switcher I see custom window names (that's where "Tab Count in Window Title" helps) and can switch between tasks/projects easily.

There are some windows/projects/contexts which I only get back to once a month. Tabs in those windows mostly remain unloaded but opened windows still feel like clutter and do consume resources on the window-manager level.

For each such rarely attended window I would love to close it and have a method of **easily** getting back to its most recent state when I need it again. I can think of two methods:

#### Sidebery snapshots
In Sidebery I can see Snapshots, and I can see numerically identified windows and I can restore windows from snapshots. All that works just fine. It doesn't feel ergonomic though. If I want to restore a window I closed a month ago, I need to remember which was the last snapshot containing the window, and which particular window inside the snapshot it is - because Sidebery's snapshots don't preserve my custom window titles (which I set with "Tab Count in Window Title" extension).

#### Firefox bookmarks
Another flow is using firefox's builtin functionality: "bookmark all tabs", "open all bookmarks". One problem with this method is that I need to delete and rebookmark-all every time I worked with the window to keep bookmarks in sync with the window's state. Also, this method is lossy: containers and tabs tree/hierarchy are lost - compared to Sidebery snapshots.

#### Better method?
My question is: what's a better method of closing a window to go back to it a few weeks later such that it's restored in its latest full state? I don't mind using an additional extension for this, but was wondering if there is a feature in Sidebery which I am missing.

**Comment by dimitry12** [comment, marked as answer]
At this point I've started using Sidebery's "Bookmarks panel" feature, which fits my flow very well.

The flow I am using is:
- take my tabs panel and use "Convert to bookmarks panel", which creates a folder with bookmarks corresponding to each tab I had
- at this point, I can open the bookmarks-subfolder as "bookmarks panel" and quickly view each bookmark, or
  - I can create a new tabs-panel and "Restore from bookmarks" - which will restore original tabs, including grouping and containers!

In other words, I've completely missed both: the concept of "bookmarks panel", and convert/restore features of Sidebery - which together allow me to do exactly what I've originally outlined.
